src/main/java/org/traccar/protocol/Gl200TextProtocolDecoder.java

```diff
@@ -210,6 +264,9 @@ private Object decodeInf(Channel channel, SocketAddress remoteAddress, String se
 
         position.set(Position.PREFIX_ADC + 1, parser.next());
         position.set(Position.PREFIX_ADC + 2, parser.next());
+        if (model.equals("GV310LAU")) {
+            position.set(Position.PREFIX_ADC + 3, parser.next());
```

Comment:
Looks like this actually breaks the implementation. The group is always present in your regex, but you're only reading it when the model is "GV310LAU". It means if the model is not "GV310LAU", the index gets messed up and everything after it is not read correctly.

Reply:
I thought .optional() in the regex would work in those cases.

Reply:
How would optional know what to skip?